cloudimg/ms_azure.py

```diff
@@ -84,7 +84,8 @@ def _get_tags_filter_expression(tags):
 
     # Create the filter expression like:
     # "tag1"='value1" and "tag2" = 'value2" ...
-    return " and ".join(["\"%s\"='%s'" % (k, v) for k, v in tags.items()])
+    items = sorted(tags.items())
+    return " and ".join(["\"%s\"='%s'" % (k, v) for k, v in items])
 
```

Comment:
Note: This change was only made to make `py27` happy during tests as it doesn't seem to respect the ordering of a `dict.update`.

Reply:
It was officially announced in python3.7 :-)
* https://docs.python.org/3/whatsnew/3.7.html
```
the insertion-order preservation nature of [dict](https://docs.python.org/3/library/stdtypes.html#typesmapping) objects [has been declared](https://mail.python.org/pipermail/python-dev/2017-December/151283.html) to be an official part of the Python language spec.
```
You would need to explicitly use `OrderedDict` in python2.7 :-)

Reply:
Thanks for letting me know.

In this case we don't necessarily need the ordered dict for the implementation to work but in order to ease the tests I had to sort the dict here.

Otherwise it's kinda difficult to `assert` the `filter_expression`  for `container_client.find_blobs_by_tags`